File: metro/src/app/position.ts
import { lonLatToCanvas } from './utils/projection';

export class Position {
  lat: number;
  lon: number;
  x: number;
  y: number;

  constructor(_width: number, _height: number, lat: number, lon: number) {
    this.lat = lat;
    this.lon = lon;
    const p = lonLatToCanvas(lon, lat);
    this.x = p.x;
    this.y = p.y;
  }
}

File: metro-sim/src/main/scala/Position.scala
// Metro. SDMT

class Position(lat: Double, lon: Double) {

  val RADIUS = 6371
  val phi1 = 40.4202961;
  val phi0 = 40.4202961;
  val lambda0: Double = -3.718762;
  val width = 3400
  val height = 2000

  val x: Double = -1 * this.RADIUS * (lon - this.lambda0) * Math.cos(this.phi1) + (width / 2)
  val y: Double = -1 * this.RADIUS * (lat - this.phi0) + (height / 2)
}

